compiler/modules/types/enumerable.bal

```diff
@@ -166,18 +166,19 @@ function compareEnumerable(EnumerableType v1, EnumerableType v2) returns Order {
         return v1 == s2 ? EQ : (v1 < s2 ? LT : GT);
     }
     else {
+        float f1 = <float>v1;
         float f2 = <float>v2;
         // JBUG: #17977 can't use `==`
-        if floatEq(v1, f2) {
+        if floatEq(f1, f2) {
             return EQ;
         }
-        else if floatEq(v1, float:NaN) {
+        else if floatEq(f1, float:NaN) {
              return LT;
         }
         else if floatEq(f2, float:NaN) {
             return GT;
         }
-        else if v1 < f2 {
+        else if f1 < f2 {
             return LT;
```

Comment:
Please don't edit this,and add another branch `else if v1 is decimal {` instead.